atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CqlCluster.java

```diff
@@ -17,29 +17,61 @@
 package com.palantir.atlasdb.cassandra.backup;
 
 import com.datastax.driver.core.Cluster;
+import com.github.rholder.retry.RetryException;
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.RangeSet;
 import com.palantir.atlasdb.cassandra.CassandraServersConfigs.CassandraServersConfig;
 import com.palantir.atlasdb.cassandra.backup.transaction.TransactionsTableInteraction;
 import com.palantir.atlasdb.keyvalue.cassandra.LightweightOppToken;
 import com.palantir.atlasdb.keyvalue.cassandra.async.client.creation.ClusterFactory;
 import com.palantir.atlasdb.keyvalue.cassandra.async.client.creation.ClusterFactory.CassandraClusterConfig;
 import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
 import java.io.Closeable;
 import java.io.IOException;
 import java.net.InetSocketAddress;
+import java.time.Duration;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
 
 public final class CqlCluster implements Closeable {
+    private static final SafeLogger log = SafeLoggerFactory.get(CqlCluster.class);
+
+    private static final Duration RETRY_DURATION = Duration.ofMinutes(1L);
+    private static final int RETRY_COUNT = 5;
+
     private final Cluster cluster;
     private final CassandraServersConfig cassandraServersConfig;
     private final Namespace namespace;
+    private final Retryer<CqlSession> cqlSessionRetryer;
 
     // VisibleForTesting
     public CqlCluster(Cluster cluster, CassandraServersConfig cassandraServersConfig, Namespace namespace) {
+        this(cluster, cassandraServersConfig, namespace, RETRY_DURATION);
+    }
+
+    // VisibleForTesting
+    public CqlCluster(
+            Cluster cluster,
+            CassandraServersConfig cassandraServersConfig,
+            Namespace namespace,
+            Duration retryDuration) {
         this.cluster = cluster;
         this.cassandraServersConfig = cassandraServersConfig;
         this.namespace = namespace;
+        this.cqlSessionRetryer = new Retryer<>(
+                StopStrategies.stopAfterAttempt(RETRY_COUNT),
+                WaitStrategies.fixedWait(retryDuration.toMillis(), TimeUnit.MILLISECONDS),
+                attempt -> !attempt.hasResult());
```

Comment:
Do we have a sense of how many connection failures happen? Do we have a metric for this?